src/script/entity/Conversation.ts

```diff
@@ -163,6 +167,7 @@ export class Conversation {
   public readonly receiptMode: ko.Observable<RECEIPT_MODE>;
   public readonly removed_from_conversation: ko.PureComputed<boolean>;
   public readonly roles: ko.Observable<Record<string, string>>;
+  public readonly isLastMessageVisible: ko.Observable<boolean>;
```

Comment:
This should not belong in the low level entity. Should be a view-only thing. 
It should leave in `Conversation.tsx` only, I believe

Reply:
It was my idea too, but then we should pass it to the `MessageList` to implement its updating, or pull all the updating logic into the `Conversation`, chaining props of `onVisible` and `onVisibilityLost`. It was quite ugly implementation. I also thought that some of the props and methods of `ConversationEntity` might not be in the perfect place (like `initialMessage`), and maybe we need a `ConversationView` service, WDYT?

Reply:
I believe we do need a cleaner architecture for those values, yes. But maybe it's not the correct PR to tackle this. Probably a follow up PR would be nice.